src/components/DiffView/index.spec.tsx

```diff
@@ -276,12 +272,55 @@ describe(__filename, () => {
     });
   });
 
+  it('does not tokenize the hunks to add syntax highlighting if the diff has been trimmed', () => {
+    const _codeCanBeHighlighted = jest.fn().mockReturnValue(true);
+    const _codeShouldBeTrimmed = jest.fn().mockReturnValue(true);
+    const _tokenize = jest.fn();
+
+    renderWithLinterProvider({
+      _codeCanBeHighlighted,
+      _codeShouldBeTrimmed,
+      _tokenize,
+      // A minified file will be trimmed by default.
+      isMinified: true,
+    });
+
+    expect(_tokenize).not.toHaveBeenCalled();
+  });
+
+  it('resets tokens to undefined if they seem invalid', () => {
+    const _codeCanBeHighlighted = jest.fn().mockReturnValue(true);
+    const _codeShouldBeTrimmed = jest.fn().mockReturnValue(false);
+    const _tokenize = (jest.spyOn(
+      ReactDiffView,
+      'tokenize',
+    ) as unknown) as typeof ReactDiffView.tokenize;
```

Comment:
We probably don't even need to spy on this function. What matters is the `tokens` prop.

Reply:
I believe we do need this. I added it because it is possible that `tokens` can be `undefined` for other reasons. We try to ensure that the test setup covers that (by having `_codeCanBeHighlighted` return `true` and `_codeShouldBeTrimmed` return `false`), but there are just enough moving parts in here that I don't think we should rely on test setup to ensure that the case does what we expect it to. If any of that logic were to change, or new code introduced, it's possible that `tokenize` might not be called at all in this test case, and that would cause `tokens` to be `undefined` and hence a false positive. If we spy on `tokenize` then we can be sure that it is actually called, and therefore this test case is valid.